### PR TITLE
Add support for 24hr-averaged ocn/ice products

### LINKED ISSUE #2674: Missing CICE hist files when cycle is not t00z and FHOUT_GFS_ICE is 24
### What is wrong?

For cases where `FHOUT_GFS_ICE` is 24 and initialized at a time that is not 00Z, CICE files are not properly linked to COMROOT. This results in missing CICE hist files in COMROOT. After investigating this issue on WCOSS2, this happens because the raw CICE output is not averaged for a full 24 hours for the first lead time. For example, if the initialization time is 12Z and FHOUT_GFS_ICE is 24, the first lead time will only be averaged from 12Z to 00Z. Subsequent lead times are averaged from 00Z to 00Z. The global-workflow predetermines that the CICE output will be averaged from 12Z to 12z, which results in the global workflow creating links that do not match the raw CICE filenames, resulting in the CICE output not to save in COMROOT. 

### What should have happened?

It may be ideal that CICE be averaged for a full 24 hours for the first lead time. For example, if the initialization time is 12Z and `FHOUT_GFS_ICE` is 24, the first lead time should be averaged from 12Z to 12Z. Subsequent lead times should be averaged from 12Z to 12Z. An alternative positive outcome would be for the global-workflow to correctly predetermine the name of the CICE output in these particular cases. 

### What machines are impacted?

WCOSS2

### Steps to reproduce

1. Set `FHOUT_GFS_ICE` to 24 in `config.base`.
2. Run a case that is not initialized at t00Z (e.g. t12z). 

### Additional information

This issue has been tested on WCOSS2, but it most likely impacts all platforms. The list of CICE output below illustrates how the global-workflow is not picking up the raw CICE output. 
link_mismatch2

### Do you have a proposed solution?

This issue has been initially addressed in PR #2561. In this PR, CICE_predet and CICE_postdet were modified so that the global-workflow can correctly predetermine the name of the raw CICE output (as illustrated in the image below). Also, the oceanice prod task was modified accordingly to account for the CICE filename changes in these cases. However, further fixes will need to be made to consider cases where `FHMIN` is not 0. 

CICE_final_output_fix

### PR DESCRIPTION
**Pull request by EricSinsky-NOAA**
# Description

This PR adds full capability to produce 24-hr ocean and ice products. In particular, this PR fixes a bug where the CICE history files are missing when `cycle` is not `t00z` and when `FHOUT_ICE` is `24`. This fix has been made possible since the changes from [UFSWM PR #2437](https://github.com/ufs-community/ufs-weather-model/pull/2437) have been merged and the GW PR #3145 updated the UFSWM hash to include these UFSWM updates for global-workflow. 

Resolves #2674 

# Type of change
- [x] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics

- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? YES
  - [ ] EMC verif-global 
  - [ ] GDAS 
  - [x] GFS-utils  [NOAA-EMC/gfs-utils#107](https://github.com/NOAA-EMC/gfs-utils/issues/107)
  - [ ] GSI 
  - [ ] GSI-monitor 
  - [ ] GSI-utils 
  - [ ] UFS-utils 
  - [ ] UFS-weather-model 
  - [ ] wxflow 

# How has this been tested?

- Clone, built and ran on WCOSS2 with the GEFS CI test

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

File: ush/parsing_namelists_CICE.sh [hunk 1]
```diff
@@ -70,7 +70,12 @@ local CICE_RESTART_FORMAT="pnetcdf2"
 local CICE_DUMPFREQ="y"  # "h","d","m" or "y" for restarts at intervals of "hours", "days", "months" or "years"
 local CICE_DUMPFREQ_N=10000  # Set this to a really large value, as cice, mom6 and cmeps restart interval is controlled by ufs.configure
 local CICE_DIAGFREQ=$(( 86400 / DT_CICE ))  # frequency of diagnostic output in timesteps, recommended for 1x per day
-local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 0, 1"
+if [[ "${RUN}" == "gefs" ]]; then
+  local CICE_DIAGFREQ1=$((( FHOUT_ICE * 3600 )/ DT_CICE ))
+  local CICE_HISTFREQ_N="0, 0, 0, ${CICE_DIAGFREQ1}, 1"
+else
+  local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 0, 1"
+fi
```
Comment (EricSinsky-NOAA) [inline ush/parsing_namelists_CICE.sh]: @NeilBarton-NOAA Will SFS have 24-hr ice products as well? If yes, I can add SFS here as well.

Comment (NeilBarton-NOAA) [reply ush/parsing_namelists_CICE.sh]: @EricSinsky-NOAA Yes, SFS is defaulting to 24 means. We may have additional monthly output, but this is fine for now

**Comment by EricSinsky-NOAA** [comment]
The GEFS 24-hr ocean and ice products were successfully created when the 12z C48 GEFS offline CI test was performed.  Marking as ready for review.

**Comment by NickSzapiro-NOAA** [comment]
It's an edge case but are timesteps always chosen to be evenly divisible into 24 hours?

fwiw, I think there's an alternate solution to this in CICE via histfreq_base, if needed

**Review by NickSzapiro-NOAA — APPROVED** [review]

File: sorc/link_workflow.sh [hunk 2]
```diff
@@ -154,7 +154,7 @@ for dir in gfs gefs sfs
 do
   ${LINK_OR_COPY} "${HOMEgfs}/sorc/upp.fd/parm/${dir}" .
 done
-for file in ice.csv ocean.csv ocnicepost.nml.jinja2; do
+for file in ice.csv ice_gefs.csv ocean.csv ocean_gefs.csv ocnicepost.nml.jinja2; do
```
Comment (aerorahul) [inline sorc/link_workflow.sh]: Should update to link `ocean_gfs.csv` and `ice_gfs.csv`

File: .gitignore [hunk 3]
```diff
@@ -74,7 +74,9 @@ parm/post/gfs
 parm/post/gefs
 parm/post/sfs
 parm/post/ocean.csv
+parm/post/ocean_gefs.csv
```
Comment (aerorahul) [inline .gitignore:77]: I think we created `ocean_gfs.csv` and `ice_gfs.csv` as well in `gfs_utils`

File: parm/post/oceanice_products_gefs.yaml [hunk 4]
```diff
@@ -10,7 +10,7 @@ ocnicepost:
     copy:
       - ["{{ EXECgfs }}/ocnicepost.x", "{{ DATA }}/"]
       - ["{{ PARMgfs }}/post/ocnicepost.nml.jinja2", "{{ DATA }}/"]
-      - ["{{ PARMgfs }}/post/{{ component }}.csv", "{{ DATA }}/"]
+      - ["{{ PARMgfs }}/post/{{ component }}_gefs.csv", "{{ DATA }}/{{ component }}.csv"]
```
Comment (aerorahul) [inline parm/post/oceanice_products_gefs.yaml:13]: should the `oceanice_products_gfs.yaml` be updated to be consistent with the changes?

**Review by aerorahul — COMMENTED** [review]
some feedback.

File: sorc/link_workflow.sh (hunk 2 above)
Comment (EricSinsky-NOAA) [reply sorc/link_workflow.sh]: @aerorahul This suggestion and your other suggestions have been committed. After some offline testing on Friday, I believe this PR is ready to move forward.

**Review by KateFriedman-NOAA — APPROVED** [review]

**Review by DavidHuber-NOAA — APPROVED** [review]

**Comment by DavidHuber-NOAA** [comment]
Relaunching CI on Gaea C6 now that the GSI fix issue has been resolved.

**Comment by EricSinsky-NOAA** [comment]
It seems the CI failure of `C96C48mx500_S2SW_cyc_gfs` is related to issue #3627.

**Comment by DavidHuber-NOAA** [comment]
@EricSinsky-NOAA Yes, I believe your are correct.  Since this is the only job that failed, I believe this PR is ready to merge.  We can address 3627 in another PR.
